stabilizer/src/main/java/com/hazelcast/stabilizer/provisioner/AwsProvisioner.java

```diff
@@ -0,0 +1,333 @@
+                addInstanceToAgentsFile(i);
+                checkedInstances.add(i);
+            }else{
+                log.warn("Time out Waiting for running status id=" + i.getInstanceId() );
+            }
+        }
+        return checkedInstances;
+    }
+
+    public void terminateInstances(int count){
+        List<AgentAddress> agents = AgentsFile.load(agentsFile);
+
+        List<AgentAddress> deadList = agents.subList(0, count);
+        agents.removeAll(deadList);
+        AgentsFile.save(agentsFile, agents);
+
+        List<Instance> deadInstances = getInstancesByPublicIp(deadList);
+
+        terminateInstances(deadInstances);
+    }
+
+
+    private void terminateInstances(List<Instance> instances){
+
+        List<String> ids = new ArrayList<String>();
+        for(Instance i : instances){
+            ids.add(i.getInstanceId());
+        }
+
+        TerminateInstancesRequest terminateInstancesRequest = new TerminateInstancesRequest();
+        terminateInstancesRequest.withInstanceIds(ids);
+        TerminateInstancesResult result = ec2.terminateInstances(terminateInstancesRequest);
+    }
+
+    public List<Instance> getInstancesFromAgentsFile(){
+        List<AgentAddress> currentAgents = AgentsFile.load(agentsFile);
+        return getInstancesByPublicIp(currentAgents);
+    }
+
+
+    public List<Instance> getInstancesByPublicIp(List<AgentAddress> agents){
+
+        List<String> ips = new ArrayList<String>();
+        for(AgentAddress agent : agents){
+            ips.add(agent.publicAddress);
+        }
+
+        DescribeInstancesRequest request = new DescribeInstancesRequest();
+
+        Filter filter = new Filter(AWS_PUBLIC_IP_FILTER, ips);
+
+        DescribeInstancesResult result = ec2.describeInstances(request.withFilters(filter));
+        List<Reservation> reservations = result.getReservations();
+
+        List<Instance> foundInstances = new ArrayList();
+
+        for (Reservation reservation : reservations) {
+            List<Instance> instances = reservation.getInstances();
+            foundInstances.addAll(instances);
+        }
+        return foundInstances;
+    }
+
+    public boolean waiteForInstanceStatusRunning(Instance i){
+
+        DescribeInstancesRequest describeInstancesRequest = new DescribeInstancesRequest().withInstanceIds(i.getInstanceId());
+        int counter=0;
+        while ( counter++ < MAX_SLEEPING_ITTERATIONS ) {
+            sleepMillis(SLEEPING_MS);
+
+            DescribeInstancesResult result = ec2.describeInstances(describeInstancesRequest);
+
+            for ( Reservation r : result.getReservations() ) {
+                for ( Instance j : r.getInstances() ) {
+                    if ( j.getPublicIpAddress() != null  && j.getState().getName().equals(AWS_RUNNING_STATE) ) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    public void addInstanceToAgentsFile(Instance i){
```

Comment:
don't use single letter variable names, unless you have some kind of counter in a for loop. In this case i would use 'instance' as name.